.github/workflows/component-builder.yml

```diff
@@ -1,62 +1,135 @@
 name: "Component Builder Check"
 on:
-  pull_request:
+  # Wait for event on pull_request_target so that this action
+  # can access repo secrets: QUAY_USER & QUAY_TOKEN
+  pull_request_target:
+    permissions:
+      contents: read
     paths:
       - 'containers/fedora/**'
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 jobs:
-  guest-fedora-amd64:
+  build-guest-fedora:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./containers/fedora
+    strategy:
+      matrix:
+        CPU_ARCH: [x86_64, aarch64, s390x]
     env:
-      FEDORA_IMAGE: Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
-      FEDORA_VERSION: 41
-      CPU_ARCH: amd64
+      CPU_ARCH: ${{ matrix.CPU_ARCH }}
+      FEDORA_VARS_FILE: "fedora-vars"
+      PR_NUMBER: ${{ github.event.number }}
       FULL_EMULATION: "true"
       DEBIAN_FRONTEND: noninteractive
     steps:
-      - name: Check out code
+      - name: Check out PR code safely (merge ref)
         uses: actions/checkout@v4
+        with:
+          ref: refs/pull/${{ env.PR_NUMBER }}/merge
       - name: Install dependencies for VM build
         run: |
           sudo apt-get update
           sudo apt-get install -y \
-            qemu-system-x86 \
             libvirt-daemon-system \
             virtinst cloud-image-utils \
-            libguestfs-tools
+            libguestfs-tools \
+            podman
+          case "${{ matrix.CPU_ARCH }}" in
+            x86_64)
+              sudo apt-get install -y qemu-system-x86
+              ;;
+            aarch64)
+              sudo apt-get install -y qemu-system-arm qemu-efi-aarch64
+              ;;
+            s390x)
+              sudo apt-get install -y qemu-system-s390x
+              ;;
+          esac
       - name: Tweak hosted runner to enable 'virt-sysprep'
         # https://bugs.launchpad.net/ubuntu/+source/linux/+bug/759725
         run: sudo chmod 0644 /boot/vmlinuz*
+      - name: Add the current user to group kvm
+        run: |
+          sudo usermod -a -G kvm $(whoami)
+          sudo chmod 0666 /dev/kvm
       - name: Fetch base Fedora image
-        working-directory: ./containers/fedora
-        run: wget -q "https://download.fedoraproject.org/pub/fedora/linux/releases/41/Cloud/x86_64/images/${{ env.FEDORA_IMAGE }}"
+        run: |
+          set -euo pipefail
+          VERS=$(sed -n "s/^FEDORA_VERSION:[[:space:]]\(.*\)/\1/pi" ${{ env.FEDORA_VARS_FILE }})
```

Comment:
not sure I follow this - don't we get the version from `FEDORA_VERSION` env variable?
should be somethig like:
```         
     - name: Set environment variables from file
            run: |
              cat <path to env file> >> $GITHUB_ENV
```